### PR TITLE
Run depopts tests jq

### PR DESCRIPTION
**Pull request by edwintorok**
Sorry for the delay. Here is take two on https://github.com/ocaml/ocaml-ci-scripts/pull/145 , using `jq` this time.

The Travis log folds may not belong in this PR, but might make it easier to read the Travis output, which is a lot longer now, and can be confusing on where each phase begins, hopefully the folds will help with that, let me know if I should make a separate PR for that.

### CODE REVIEW

**Comment by samoht** [comment]
That looks pretty great but it seems that the fold doesn't work? Maybe the CI is not testing the right yorick.ml? Could you try to add a temporary commit which point to your fork just to be sure?

Also, did you try to test it on packages which failed previously?

**Comment by dsheets** [comment]
The problem is #84, I believe.

**Comment by edwintorok** [comment]
I'll setup a fork, run it through Travis and report back.

**Comment by edwintorok** [comment]
Fixed two problems:
  - `jq` query was returning empty packages called `.`
  - fixed folding: spaces are not supported in the fold name (that is displayed on the right), and had to flush the printer.

Its all green now on my fork:
https://travis-ci.org/edwintorok/datakit
https://travis-ci.org/edwintorok/ocaml-git

If you want to see the folds (it appears to be done in javascript, takes a while to get the whole log loaded and then folded):
https://travis-ci.org/edwintorok/datakit/jobs/190690365
https://travis-ci.org/edwintorok/ocaml-git/jobs/190676453

**Comment by samoht** [comment]
That looks pretty good, thanks! The fold output could be improved later on, I think this is good enough for merging. @dsheets maybe you want to review first?

**Comment by samoht** [comment]
ping @dsheets. I am pretty keen to merge this and revert if it breaks, I'll do that tomorrow.

**Comment by dsheets** [comment]
Oops! I missed your first ping, sorry. Taking a look now.

File: ci_opam.ml [hunk 1]
```diff
@@ -118,7 +118,12 @@ let install ?(depopts="") ?(tests=false) args =
 let install_with_depopts depopts =
   install ~tests:tests_run ~depopts ["-v"];
   ?|~ "opam remove %s -v" pkg;
-  ?|~ "opam remove -a %s" (filter_base depopts)
+  ?|~ "opam remove -a %s" (filter_base depopts);
+  if tests_run then begin
+    install ~tests:false ~depopts ["-v"];
+    ?|~ "opam remove %s -v" pkg;
+    ?|~ "opam remove -a %s" (filter_base depopts);
+  end
```
Comment (dsheets) [inline ci_opam.ml]: I don't understand the purpose of this commit. It seems vaguely logical but still seems confusing. Perhaps add a comment describing why this is done?

**Comment by dsheets** [comment]
This all looks really excellent. I just have one question (above) about 28f19f8.

File: ci_opam.ml (hunk 1 above)
Comment (edwintorok) [inline ci_opam.ml]: The idea is that dependencies may be specified wrongly in opam, e.g. a dependency is specified as test-only, whereas some .ml file would need it during a regular build. Without this commit Travis would only do a test with installing every dep, succeed, then a user  tries to install it and it breaks. With this commit it tests both scenarios (install test-only deps+run-test, install just build deps + just build)

**Comment by edwintorok** [comment]
Thanks for the review, I added a comment in the commit and force pushed the branch.

**Comment by samoht** [comment]
Thanks for the patch and the review, merging!

**Pull request merged**